src/goto-programs/remove_virtual_functions.cpp

```diff
@@ -322,13 +384,99 @@ void remove_virtual_functionst::get_functions(
 
   // iterate over all children, transitively
   std::set<irep_idt> visited;
-  get_child_functions_rec(
+  get_child_functions(
     class_id,
     root_function.symbol_expr,
     component_name,
     functions,
     visited);
 
+  const auto find_empty = [&](const dispatch_table_entryt &entry) {
+    return entry.symbol_expr == symbol_exprt();
+  };
+  bool empty_found =
+    std::any_of(functions.begin(), functions.end(), find_empty);
+
+  if(empty_found && !functions.empty())
+  {
+    std::cout << "problematic dispatch table\n";
+    for(const auto &f : functions)
+      std::cout << f.class_id << ":" << f.symbol_expr.pretty() << " ";
+    std::cout << std::endl;
+
+    // with a non-empty virtual dispatch table but only empty function
+    // expressions, we could not find a class that provides the required
+    // method. Therefore we try finding it in the set of parent classes of a
+    // class in the dispatch table.
+
+    std::map<irep_idt, symbol_exprt> new_entries;
```

Comment:
Why are we doing this as a fixup step, not getting it right the first time? During the walk over children I could simply ask at each node, if there is no definition at this child, use resolve-concrete to search for an inherited definition